cmd/clusterctl/client/cluster/upgrader.go

```diff
@@ -143,6 +149,13 @@ func (u *providerUpgrader) ApplyPlan(coreProvider clusterctlv1.Provider, contrac
 	log := logf.Log
 	log.Info("Performing upgrade...")
 
+	// Check for multiple instances of the same provider for v1alpha4 contract.
+	if contract == v1alpha4Contract {
```

Comment:
Should this be `contract > v1alpha3`? Is there api version parsing library that can be used for this?

Reply:
If we are moving this check into the `doUpgrade` func, then we can rely on the upgrade plan and do check only if the current contract is v1alpha3

Reply:
I wasn't able to find the current contract in upgradePlan, I could only find `upgradePlan.contract` which I believe is the contract that we want to upgrade to.